### ISSUE TITLE
Port forwarding broken for helm releases

### ISSUE THREAD

**Issue by tdickman** [GoogleContainerTools/skaffold#2379]
### Expected behavior

Port forwarding works properly for helm releases.

### Actual behavior

Port forwarding does not occur for helm releases

### Information

- Skaffold version: [latest version](https://github.com/GoogleContainerTools/skaffold/commit/734e5e3aee4611445237fd34f1abd03e36fdfb64) built from master
- Operating system: ubuntu
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta11
kind: Config
build:
  tagPolicy:
    sha256: {}
  artifacts:
  - image: gcr.io/k8s-skaffold/skaffold-helm
deploy:
  helm:
    releases:
    - name: skaffold-helm
      chartPath: skaffold-helm
      #wait: true
      #valuesFiles:
      #- helm-skaffold-values.yaml
      values:
        image: gcr.io/k8s-skaffold/skaffold-helm
      #recreatePods will pass --recreate-pods to helm upgrade
      #recreatePods: true
      #overrides builds an override values.yaml file to run with the helm deploy
      #overrides:
      # some:
      #   key: someValue
      #setValues get appended to the helm deploy with --set.
      #setValues:
        #some.key: someValue
```

### Steps to reproduce the behavior

1. Run `skaffold dev` with [helm-deployment example](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/helm-deployment) using skaffold built from HEAD
2. No port forward occurs

cc/ @priyawadhwa since this might be related to #2215

**Comment by priyawadhwa** [comment]
Hey @tdickman I just looked into it -- looks like automatic port forwarding of services isn't working with helm because we don't add a skaffold label after deployment (the port forwarding determines which services to port forward by filtering by this label).

I'll look into adding the label post helm deploy, otherwise user defined port forwarding does still work as exptected (I added 

```yaml
portForward:
- resourceType: deployment
  resourceName: skaffold-helm
  port: 80
  localPort: 8080
```

to my skaffold.yaml with the `helm-deployment` example.

**Comment by priyawadhwa** [comment]
blocked on #2142

**Comment by tdickman** [comment]
> blocked on #2142

Not sure this needs to be blocked on helm 3 support since helm 3 is still prerelease.

**Comment by priyawadhwa** [comment]
Hey @tdickman, we're thinking of using the `helm template` subcommand and then adding the necessary label to the resources in the output to fix this issue. Before we choose to rely on that subcommand, helm 3 would have to be more mature, so we would want #2142 to be completed first.

**Comment by tdickman** [comment]
Helm 3 is still in alpha, so I would guess it's gong to be a while until it is out and well supported (this could very well be 6 months away). I'm not sure it makes sense to block on this.

Also interested to discuss the use of `helm template` - that command currently exists in helm 2, any specific reason you are moving to it / waiting on helm 3?

**Comment by priyawadhwa** [comment]
For sure, so we actually had a PR open to fix this issue before (#2105), and the PR was ultimately [closed](https://github.com/GoogleContainerTools/skaffold/pull/2105#issuecomment-493189739) with the intention of waiting for helm 3. The discussion in that PR should answer your questions, but if not please let us know!

I think ultimately the issue was that patching live resources resulted in redeployments, which is not an ideal experience. Patching the resources before deployment was also considered, but right now there isn't a good way of determining the output yamls before helm deploys them (as described in this [comment](https://github.com/GoogleContainerTools/skaffold/pull/2075#issuecomment-507548893) here). 

cc @tejal29, this is the current state, correct?

**Comment by zodiac1214** [comment]
> Hey @tdickman I just looked into it -- looks like automatic port forwarding of services isn't working with helm because we don't add a skaffold label after deployment (the port forwarding determines which services to port forward by filtering by this label).
> 
> I'll look into adding the label post helm deploy, otherwise user defined port forwarding does still work as exptected (I added
> 
> ```yaml
> portForward:
> - resourceType: deployment
>   resourceName: skaffold-helm
>   port: 80
>   localPort: 8080
> ```
> 
> to my skaffold.yaml with the `helm-deployment` example.

I tried this with my own helm chart it doesn't work :(

`kubeclt port-forward TYPE/NAME -n NAMESPACE` works fine for me

**Comment by balopat** [comment]
I do think this is pretty important to support with helm 2 as well, and has high priority as such given that we have a lot of users depending on it. Helm 2's `helm template` might be a viable solution now and that would be a benefit for other features too. @tejal29 do you want to give #2105 another go?

**Comment by tejal29** [comment]
@priyawadhwa , since we are going to add new helm-template deployer which will use the **Manifest Labeller for helm** mentioned in #2105, i am adding a fix here to rely on "release=<release-name>" label that helm applies to all the resource deployed via `helm install`
See https://github.com/GoogleContainerTools/skaffold/pull/2557
On my branch
```
@helm-deployment (fix_2379)$ skaffold dev --port-forward=true
Generating tags...
 - gcr.io/k8s-skaffold/skaffold-helm -> gcr.io/k8s-skaffold/skaffold-helm:latest
Tags generated in 110.294µs
Starting build...
Found [gke_tejal-test_us-central1-a_dump] context, using local docker daemon.
Building [gcr.io/k8s-skaffold/skaffold-helm]...
Sending build context to Docker daemon  4.096kB
Step 1/2 : FROM nginx:stable
 ---> 3014a6ac1176
Step 2/2 : COPY static /usr/share/nginx/html/
 ---> Using cache
 ---> e3c1f7501fa3
Successfully built e3c1f7501fa3
Successfully tagged gcr.io/k8s-skaffold/skaffold-helm:latest
Build complete in 159.043255ms
Starting test...
Test complete in 12.294µs
Starting deploy...
Error: release: "skaffold-helm" not found
Helm release skaffold-helm not installed. Installing...
No requirements found in skaffold-helm/charts.
NAME:   skaffold-helm
LAST DEPLOYED: Fri Jul 26 10:24:04 2019
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/Ingress
NAME                         HOSTS  ADDRESS  PORTS  AGE
skaffold-helm-skaffold-helm  *      80       0s

==> v1/Pod(related)
NAME                           READY  STATUS             RESTARTS  AGE
skaffold-helm-7564fd45f-h6dbx  0/1    ContainerCreating  0         0s

==> v1/Service
NAME                         TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)  AGE
skaffold-helm-skaffold-helm  ClusterIP  10.23.250.144  <none>       80/TCP   1s

==> v1beta1/Deployment
NAME           DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
skaffold-helm  1        1        1           0          0s


Deploy complete in 2.932925744s
Watching for changes...

Port forwarded service/skaffold-helm-skaffold-helm from remote port 80 to local port 4503
```

**Comment by tejal29** [comment]
So, @priyawadhwa i chose a wrong example. 
The `helm-deployment` example add the label `{{.ReleaseName}}`
```
tejaldesai:skaffold (gke_load_balancer_svc)$ cat examples/helm-deployment/skaffold-helm/templates/deployment.yaml 
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: {{ .Release.Name }}
  labels:
    app: {{ template "skaffold-helm.name" . }}
    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
spec:
  replicas: {{ .Values.replicaCount }}
  template:
    metadata:
      labels:
        app: {{ template "skaffold-helm.name" . }}
        release: {{ .Release.Name }}
    spec:
      containers:
....
tejaldesai:skaffold (gke_load_balancer_svc)$
```
So the solution in PR #2557 wont work.
I opened another PR # where in we bring back adding labels to services.
